azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java

```
@@ -773,6 +776,205 @@ private Connection getConnection() throws ExecutorManagerException {
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see azkaban.executor.ExecutorLoader#addExecutor(java.lang.String, int)
+   */
+  @Override
+  public Executor addExecutor(String host, int port)
+    throws ExecutorManagerException {
+    // verify, if executor already exists
+    Executor executor = fetchExecutor(host, port);
+    if (executor != null) {
+      throw new ExecutorManagerException(String.format(
+        "Executor %s:%d already exist", host, port));
+    }
+    // add new executor
+    addExecutorHelper(host, port);
+    // fetch newly added executor
+    executor = fetchExecutor(host, port);
+
+    return executor;
+  }
+
+  private void addExecutorHelper(String host, int port)
+    throws ExecutorManagerException {
+    final String INSERT =
+      "INSERT INTO executors " + "(host, port) values (?,?)";
+    QueryRunner runner = createQueryRunner();
+    String errString = String.format("Error adding %s:%d ", host, port);
+    try {
+      int rows = runner.update(INSERT, host, port);
+      if (rows == 0) {
+        throw new ExecutorManagerException(errString);
+      }
+    } catch (SQLException e) {
+      throw new ExecutorManagerException(errString, e);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see azkaban.executor.ExecutorLoader#postEvent(azkaban.executor.Executor,
+   *      azkaban.executor.ExecutorLogEvent.EventType, java.lang.String,
+   *      java.lang.String)
+   */
+  @Override
+  public boolean postEvent(Executor executor, EventType type, String user,
+    String message) {
+    QueryRunner runner = createQueryRunner();
+
+    final String INSERT_PROJECT_EVENTS =
+      "INSERT INTO executor_events (executor_id, event_type, event_time, username, message) values (?,?,?,?,?)";
+    Date updateDate = new Date();
+    try {
+      runner.update(INSERT_PROJECT_EVENTS, executor.getId(), type.getNumVal(),
+        updateDate, user, message);
+    } catch (SQLException e) {
+      e.printStackTrace();
```

better to throw an exception